src/configurationProvider.ts

```diff
@@ -288,6 +346,31 @@ function convertLogLevel(commonLogLevel: string) {
 }
 
 interface IMainClassOption {
-    readonly projectName?: string;
     readonly mainClass: string;
+    readonly projectName?: string;
+    readonly filePath?: string;
+}
+
+interface IMainClassQuickPickItem extends vscode.QuickPickItem {
+    item: IMainClassOption;
+}
+
+class MostRecentlyUsedHistory {
+    private cache: { [key: string]: number } = {};
+
+    public getMRUTimestamp(mainClassOption: IMainClassOption): number {
+        return this.cache[this.getId(mainClassOption)] || 0;
+    }
+
+    public updateMRUTimestamp(mainClassOption: IMainClassOption): void {
+        this.cache[this.getId(mainClassOption)] = Date.now();
+    }
+
+    public contains(mainClassOption: IMainClassOption): boolean {
+        return Boolean(this.cache[this.getId(mainClassOption)]);
+    }
+
+    private getId(mainClassOption: IMainClassOption): string {
```

Comment:
getKey to be consistent with the cache property name?

Reply:
fixed.